backend/src/index.js

```diff
@@ -52,50 +54,48 @@ io.on( 'connection', socket => {
 			issueUser.sockets.push( socket.id );
 		}
 
-		issueUser.tabs.set( socket.id, message.user.state );
+		issueUser.tabs[ socket.id ] = message.user.state;
 
 		issueUser.user = message.user;
 
 		const users = getUsers( issueUsers );
 
+		await storeUsers( issueKey, JSON.stringify( issueUsers ) );
+
 		socket.broadcast.to( issueKey ).emit( 'refresh', users );
 
 		reply( null, users );
 	} );
 
-	socket.on( 'removeUser', message => {
-		const issueKey = `${ message.repoName }:${ message.issueId }`;
+	socket.on( 'removeUser', async message => {
+		const issueKey = createIssueKey( message.repoName, message.issueId );
 
-		removeUser( issueKey );
+		await removeUser( issueKey );
 	} );
 
-	socket.on( 'disconnect', () => {
+	socket.on( 'disconnect', async () => {
 		for ( const issueKey of socket.session.issues || [] ) {
-			removeUser( issueKey );
+			await removeUser( issueKey );
 		}
 
 		delete socket.issues;
 	} );
 
-	function removeUser( issueKey ) {
-		const issueUsers = db.get( issueKey );
+	async function removeUser( issueKey ) {
+		const issueUsers = JSON.parse( await fetchUsers( issueKey ) || '{}' );
 
-		if ( !issueUsers ) {
+		if ( Object.keys( issueUsers ).length === 0 ) {
```

Comment:
There's no need for it. The `issueUser` will be undefined in the next line and the fn will return anyway.

Reply:
Good point. Resolved.